### ISSUE TITLE
Lumen - Registering Class in Service Provider

### ISSUE THREAD

**Issue by memoishin** [goaop/goaop-laravel-bridge#14]
I am registering an Aspect class in the AopServiceProvider. However, it does not seem to work.

Here is my code:

```
public function register()
{
     $this->app->singleton(LoggingAspect::class, function (Application $app) {
        return new LoggingAspect($app->make(LoggerInterface::class));
    });

    $this->app->tag([LoggingAspect::class], 'goaop.aspect');
}
```
Now even if I have the code as this:

```
public function register()
{
     $this->app->singleton(LogsasasagingAspect::class, function (Application $app) {
        return new LoggingAspect($app->make(LoggesasarInterface::class));
    });

    $this->app->tag([LoggasaingAspect::class], 'goaop.aspect');
}
```
It still doesn't show any error.

This service provider has been registered as:

`$app->register(App\Providers\AopServiceProvider::class);`
Is this normal or is there any fix for this?

**Comment by SerafimArts** [comment]
Registration of services in a DI Container is declarative and should not cause errors until the specified service is resolved.

In this case, errors will occur when u call: `$app->make(LogsasasagingAspect::class)`.

But I did not understand the question. :D

**Comment by memoishin** [comment]
As you mentioned,  "errors will occur when u call: $app->make(LogsasasagingAspect::class)", however, i do not receive any error message.

Also, the aop functionality does not work. I am using Lumen 5.6. Can this be the reason why it doesn't work?

**Comment by SerafimArts** [comment]
To be honest, I did not check the package works for Lumen at all. Here you need the experience of other developers who have adapted it.

Ping @lisachenko

**Comment by lisachenko** [comment]
I think @qrazi has an experience with Lumen, because he have contributed #11

**Comment by memoishin** [comment]
I have managed to make it work using some concepts from the slides by Chris Flynn and using the demo examples that was provided.

I believed making this change in bootstrap/app.php made it work:

use Go\Aop\Features;
$applicationAspectKernel = App\lib\ApplicationAspectKernel::getInstance();

$applicationAspectKernel->init([
    'debug'    => true,
    'appDir'   => __DIR__ . '/../app/Http/Controllers/',
    'cacheDir' => __DIR__ . '/../cache',

    'features' => Features::INTERCEPT_FUNCTIONS,
]);

**Comment by qrazi** [comment]
Right, so obviously I didn't finish testing after commit #11... 😖 Based on this issue testing the lumen integration and:

- Lumen has no `config_path` helper function, but bridge package uses it
- https://github.com/laravel/lumen-framework/issues/607 : i.e. if the bridge is installed based on the docs, the bridge package loads all registered Aspect classes before any Aspect classes are registered by other providers

https://github.com/laravel/lumen-framework/pull/723/files should allow Lumen to have a more Laravel-like registration of ServiceProviders, i.e. the `boot()` of each provider will only be called after the `register()` of all providers has run.

I'll look more into this to get it into a proper solutions, leaving this issue open
